varinfo/umm_var.py

```diff
@@ -346,53 +346,74 @@ def get_json_serializable_value(input_value: Any) -> Any:
-    For a successful requests all of these fields must be entered
+
+        For a successful requests all of these fields must be entered
 
     """
     # Required UMM-Var headers for ingesting variable entries
     headers_umm_var = {
         'Content-type': 'application/vnd.nasa.cmr.umm+json;version='
         + f'{umm_var_dict["MetadataSpecification"]["Version"]}',
         'Authorization': auth_header,
-        'Accept': 'application/json'}
+        'Accept': 'application/json'
+    }
+    variable_native_id = generate_variable_native_id(collection_id,
+                                                     umm_var_dict)
 
     url_endpoint = (cmr_env.replace('search', 'ingest') + 'collections/'
-                    f'{collection_id}/variables/{umm_var_dict["Name"]}')
+                    f'{collection_id}/variables/{variable_native_id}')
 
     response = requests.put(url_endpoint,
                             json=umm_var_dict,
                             headers=headers_umm_var,
                             timeout=10)
 
-    # A successful request returns the variable concept-id
-    # e.g.'V1259791517-EEDTEST'
     if response.ok:
+        # A successful request returns the variable concept-id
+        # e.g., 'V1259791517-EEDTEST'
         return response.json()['concept-id']
-    # A failed request returns the response dict with the error message
-    # e.g. {'errors': ['#: required key [LongName] not found']}
-    return response.json()
+    else:
+        # A failed request returns the response dict with the error message
+        # e.g., {'errors': ['#: required key [LongName] not found']}
+        # This will be converted into a single string that can be returned to
+        # the end-user.
+        return '\n  '.join(response.json()['errors'])
 
 
 def publish_all_umm_var(collection_id: str,
                         all_umm_var_dict: Dict,
                         auth_header: str,
-                        cmr_env: CMR_UAT = CMR_UAT) -> Dict:
+                        cmr_env: CmrEnvType = CMR_UAT) -> Dict[str, str]:
     """ Publish all UMM-Var entries associated with a collection to CMR given:
         * collection_id: a collection's concept_id
         * all_umm_var_dict: a nested dictionary containing
             dictionaries of all UMM-Var entries for a collection
-        * auth_header: Earthdata Login (EDL) bearer_token or LaunchPad token
-            with their respective headers
+        * auth_header: Authorization HTTP header, containing a LaunchPad
+          token: 'Authorization: <token>'
         * cmr_env: CMR environments (OPS, UAT, and SIT)
     For a successful requests all of these fields must be entered
     """
     return {
         var_name: publish_umm_var(collection_id, umm_var, auth_header, cmr_env)
-        for var_name, umm_var in all_umm_var_dict.items()}
+        for var_name, umm_var in all_umm_var_dict.items()
+    }
+
+
+def generate_variable_native_id(collection_concept_id: str,
```

Comment:
Oh whoops I think you misunderstood me, I just meant can you switch the order of 'generate_variable_native_id` and `publish_umm_var` so it comes first. Since you are using 'generate_variable_id` in `publish_umm_var`. By "defining" i just meant where you literally define the function not adding a doc string. Sorry that was unclear.
So this:
```
def generate_variable_native_id(....)
....
def publish_umm_var(...)
```

Reply:
Ohhhh. I understand now. I think I'm going to push back on this one. I can do that, sure.

Reply:
Not sure if you are or not... In any case that was what I was talking about in `generate_umm_var.is_variable_concept_id` when I said:
 > Also do you mind defining is_variable_concept_id before using it?

Reply:
Ha - yeah sorry for the confusion! I moved `generate_variable_native_id` up and I think I pushed that commit. I made a comment on the `is_variable_concept_id` and we can track that conversation separately, if that sounds good.